deploy-contract/README.md

```diff
@@ -1,63 +1,829 @@
-# Bodhi.js Example - Contract Interaction
-This is a basic example that demonstrates how to sign EVM+ transactions with a polkadot wallet and [bodhi.js](https://github.com/AcalaNetwork/bodhi.js/).
+# Bodhi.js example: Contract Interaction
 
-app: https://bodhi-example-contract.vercel.app/
+A basic example on how to sign Acala EVM+ transactions with a Polkadot vallet and the [bodhi.js](https://github.com/AcalaNetwork/bodhi.js/) dependency.
 
-## Run
-- start a dev server: `yarn dev`
-- build: `yarn build`
-- lint: `yarn lint`
+**INFO: You can find a hosted version of this example here: [https://bodhi-example-contract.vercel.app/](https://bodhi-example-contract.vercel.app/)**
 
-## Core Flow
-There are 4 steps 
+## Setup
 
-**step 1**: we first need to connect a bodhi signer provider to chain node
-```tsx
+In order to be able to focus on the benefits of using `bodhi.js` in the dApp development, we will be using `vite` build tool and its `react-ts` template:
+
+````shell
+yarn create vite deploy-contract --template react-ts
+````
+
+This will create a `deploy-contract` directory with a simple `vite` template with TypeScript support.
+
+With the skeleton of our dApp ready, it can be modified to use `bodhi.js` to interact with the Acala EVM+. In order to be able to use it in the dApp, it needs to be added to the project:
+
+````shell
+yarn add @acala-network/bodhi
+````
+
+Additionally `@polkadot/extension-dapp` is needed for the dApp to be able to retrieve all of the providers added to the page:
+
+````shell
+yarn add @polkadot/extension-dapp
+````
+
+To have easy access to the components for the dApp, `antd` is required. Styling is handled by `sass` and `eslit` is used for linting:
+
+````shell
+yarn add antd
+yarn add --dev sass eslint
+````
+
+`eslint` needs to be initiated:
+
+````shell
+yarn eslint --init
+````
+
+The example project uses the following options for `eslit` initialization:
+
+````shell
+Need to install the following packages:
+  @eslint/create-config
+Ok to proceed? (y) y
+✔ How would you like to use ESLint? · problems
+✔ What type of modules does your project use? · esm
+✔ Which framework does your project use? · react
+✔ Does your project use TypeScript? · No / Yes
+✔ Where does your code run? · browser
+✔ What format do you want your config file to be in? · JavaScript
+The config that you've selected requires the following dependencies:
+
+eslint-plugin-react@latest @typescript-eslint/eslint-plugin@latest @typescript-eslint/parser@latest
+✔ Would you like to install them now? · No / Yes
+✔ Which package manager do you want to use? · yarn
+````
+
+To use the same linting configuration, replace the autogenerated `.eslintrc.cjs` with the [`eslintrc.js`](./.eslintrc.js).
+
+Once the setup is complete, add the build target to the [`vite.config.ts`](./vite.config.ts):
+
+````typescript
```

Comment:
isn't this part auto generated?

Reply:
`vite.config.ts` is autogenerated, but the build target had to be added. It was in your example, but it wasn't in my parallel one. Once I added the build target to mine, `yarn build` finally managed to build the project